src/github.com/getlantern/idletiming/idletiming_listener.go

```diff
@@ -33,9 +32,13 @@ type idleTimingListener struct {
 func (l *idleTimingListener) Accept() (c net.Conn, err error) {
 	c, err = l.orig.Accept()
 	if err == nil {
-		c = Conn(c, l.idleTimeout, func() {
-			l.onIdle(c)
-		})
+		var onIdle func()
+		if l.onIdle != nil {
+			onIdle = func() {
+				l.onIdle(c)
+			}
```

Comment:
No need, Conn already allows onIdle to be nil.  It's part of the contract for Conn.